UKHO.ExternalNotificationService.API/UKHO.ExternalNotificationService.API/Controllers/WebhookController.cs

```diff
@@ -44,10 +48,38 @@ public virtual async Task<IActionResult> Post()
         {
             using var reader = new StreamReader(Request.Body, Encoding.UTF8);
             string jsonContent = await reader.ReadToEndAsync();
+            CustomCloudEvent customCloudEvent = JsonConvert.DeserializeObject<CustomCloudEvent>(jsonContent);
 
-            CustomEventGridEvent eventGridEvent = JsonConvert.DeserializeObject<CustomEventGridEvent>(jsonContent);
+            _logger.LogInformation(EventIds.ENSWebhookRequestStart.ToEventId(), "External notification service webhook request started for event:{eventGridEvent} and _X-Correlation-ID:{correlationId}.", JsonConvert.SerializeObject(customCloudEvent), GetCurrentCorrelationId());
 
-            _logger.LogInformation(EventIds.ENSWebhookRequestStart.ToEventId(), "External notification service webhook request started for event:{eventGridEvent} and _X-Correlation-ID:{correlationId}.", JsonConvert.SerializeObject(eventGridEvent), GetCurrentCorrelationId());
+            if (customCloudEvent.Type != null)
```

Comment:
This looks an unreachable code, JsonConvert.DeserializeObject won't return null any time (it may throw exception though). 
We have compared the request with the type of event we received.